feat(.github/ISSUE_TEMPLATE): add bug and feature templates

Wabri opened the pull request:
# Description

Since the contributions are increasing it might be helpful to have a templates to apply a bug report or feature request.

If this pr will be merged when someone want to open a pr can choose between:

- bug report
- feature request
- blank issue (by config we can also remove this possibility)

![Screenshot 2024-08-05 at 18 06 26](https://github.com/user-attachments/assets/ff995ef6-79df-4649-88d8-2d28c5701cf7)

If for example the user want to apply a **bug report** this is the form that he/she/they have:

![image](https://github.com/user-attachments/assets/23169a19-a007-40c9-9c5e-1abc5dc7304b)

He/she/they have to fill out all the required elements in order to submit.

# How to try the templates

Just use my fork: https://github.com/Wabri/community.sap_install/issues/new/choose

# Example of submit

You can find at this link => https://github.com/Wabri/community.sap_install/issues/142

# Additional infos

To start using it you will need to merge to main.

# Question

Can be useful also for the other repositories?

berndfinger commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,114 @@
+      options:
+        - label: I've read [the documentation](https://github.com/sap-linuxlab/community.sap_install/tree/main/docs#readme).
+          required: true
+        - label: I've already check for existing duplicated [issues](https://github.com/sap-linuxlab/community.sap_install/issues?q=is%3Aissue+label%3Abug).
```
grammar: check -> checked

My proposals:
"I have already searched for existing duplicate issues and found none"
Alternative: 
"I have already searched for existing duplicate issues." plus:
- [checkbox] I found none.
- [checkbox] I found the following which might be related (add links to issues): [free text entry field]

Wabri replied:
Why does someone need to open a new issue instead of commenting on the existing one?

berndfinger replied:
My point was that users should be urged to search for existing issues and then report if they found none or if they found possibly related issues. In the latter case, they may of course decide not to open a new issue. Or they believe that their issue is not completely covered by an existing issue.

Wabri replied:
Make sense, I've updated the template with a text area, see https://github.com/sap-linuxlab/community.sap_install/pull/831/commits/59842f557f3b3571fd4005eaa3dd4592253879d4

berndfinger commented on .github/ISSUE_TEMPLATE/feature_request.yml:
```diff
@@ -0,0 +1,76 @@
+      description: Do you want to implement this feature by yourself?
+      options:
+        - I can implement this feature
+        - I can't implement by myself this feature
```
word order:
I can't/won't implement this feature by myself

berndfinger commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,114 @@
+    id: additional
+    attributes:
+      label: Additional information
+      description: Do you intend to submit a pr to solve this bug?
```
I suggest to use "PR" or "pull request" here. For also supporting entries by new or inexperienced GitHub users, I prefer "pull request".

Wabri replied:
You are right!

Wabri commented:
## Title standardisation through automation 

I've added an automation that fill the title by following this standard:

```
[Bug] <role_name> on <os_family>: <short_description>
```

and 
```
[Feature] <role_name> on <os_family>: <short_description>
```

The role name and os family is written only if applicable.

## Checks

For now I've added a couple of checks for the titles:

1. the object of the title `[Bug]` or `[Feature]` should be link with labels `bug` or `enhancement`. Example (see the comment in the issue linked): https://github.com/Wabri/community.sap_install/issues/141
2. the title should include a short description. Example (see the comment in the issue linked): https://github.com/Wabri/community.sap_install/issues/139

## Title update itself

Once the user create a issue that follow the standard, the automation will fill with the information found in the issue description. For example an issue with this title `[Bug]: something dont work` become `[Bug] sap_ha_pacemaker_cluster on RHEL: something dont work` with sap_ha_pacemaker_cluster taken from the issue description same for RHEL.

Here an example: https://github.com/Wabri/community.sap_install/issues/140

Wabri commented:
If you have suggestions and improvements just ping me.

berndfinger reviewed (approved):
LGTM - I think we should give it a try!
